### PR TITLE
Feature/list running images and stop them

### PR DESCRIPTION
**Pull request by axlgithub**

### CODE REVIEW

File: src/PharoLauncher-CLI-Tests/PhLImageCopyCliCommandTest.class.st [hunk 1]
```diff
@@ -15,6 +15,17 @@ PhLImageCopyCliCommandTest >> testCopyImageNameAlreadyUsed [
 	self assert: self imageRepository imagesName equals: #('myImage2' 'myImage1' ) asSet. 
 ]
 
+{ #category : #tests }
```
Comment (demarey) [inline src/PharoLauncher-CLI-Tests/PhLImageCopyCliCommandTest.class.st]: does not look related to the feature

File: src/PharoLauncher-CLI/PhLImageCopyCliCommand.class.st [hunk 2]
```diff
@@ -49,11 +49,13 @@ PhLImageCopyCliCommand class >> launcherCmdPositionals [
...
 { #category : #'command execution' }
 PhLImageCopyCliCommand >> executeWithImage: anImage [
-	[ self imageRepository copyImage: anImage to: self targetName ]
+	[[ self imageRepository copyImage: anImage to: self targetName ]
 		on: PhLNameNotAvailableError
 		do: [ ^ self errorStream
 				nextPutAll: self targetName,'Name already used';
-				newLine ]
+				newLine ]]
+		on: MessageNotUnderstood
```
Comment (demarey) [inline src/PharoLauncher-CLI/PhLImageCopyCliCommand.class.st]: does not look related to the feature + we should not rely on MessageNotUnderstood for business code

File: src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		description: 'kill all the running pharo images ';
+		meaning: [ :arg :app | true ];
+		implicitMeaning: [ :arg :app | false ])}.
+]
+
+{ #category : #'command line - arguments' }
+PhLImageKillCliCommand class >> launcherCmdPositionals [
+	^ {
+		((ClapPositional id: #existingImageName)
+				description: 'New image.';
```
Comment (demarey) [inline src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st]: description does not reflect the meaning of the argument

File: src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+PhLImageKillCliCommand >> execute [
+	(arguments at: #all) value
+		ifTrue: [ ^ LibC
+				runCommand:
+					'kill $(ps x | grep  -i Pharo | grep ''.image'' | tr -s '' '' | cut -d '' '' -f1)' ].
+	LibC
+		runCommand:
+			'kill $(ps x | grep  -i Pharo | grep ''.image''  | grep '
+				, self imageName , ' | tr -s '' '' | cut -d '' '' -f1)'
+
```
Comment (demarey) [inline src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st:49]: It would be nice to embed both Libc calls in a dedicated method: #killAllImages #killImage: imageName

File: src/PharoLauncher-CLI/PhLImageProcessListCliCommand.class.st [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+]
+
+{ #category : #'command line' }
+PhLImageProcessListCliCommand class >> launcherCmdDescription [
+	^ 'Lists all running images on local computer.'
+]
+
+{ #category : #'command execution' }
+PhLImageProcessListCliCommand >> execute [
+	"The brackets between the P of Pharo is to avoid to list the get procress whn using ps x."
```
Comment (demarey) [inline src/PharoLauncher-CLI/PhLImageProcessListCliCommand.class.st]: typo whn => when

**Comment by Bajger** [comment]
Bash command line unit tests need more changes (I will provide more commits to PR, so will let you know). 
But when I tested manually from cmd line, kill command and process list seems to be working..
My further catch-ups: 
- Way of obtaining VM path and launcher image name is quite challenge (obviously doesn't work on my local environment). 
Can we use env. variables for launcher VM path and its image path (that would be assigned when installed launcher)? Would it mean then sudo installation?  
- There is also wrong parameter to VM (instead of --nodisplay should be --headless).   
-  Print outputs of process list is not straightforward. I will make separate issue for that.  
- script `ensureShunitIsPresent.sh` always downloads zip and extracts to local directory (even if installed previously by first unit test). Need to change it, so that another unit test .sh will not download always the same thing. (will submit to this PR)

File: test/testKillImage.sh [hunk 6]
```diff
@@ -5,7 +5,7 @@
 
 launchCommand="./pharo-launcher.sh image launch ./../shared/PharoLauncher.image"
 killCommand="./pharo-launcher.sh image kill --all"
-processListCommand="./pharo-launcher.sh image ps"
+processListCommand="./pharo-launcher.sh image processList"
```
Comment (demarey) [inline test/testKillImage.sh]: We choose `ps` to "mimick" Docker API `docker ps` that list docker containers. 
`processList` is more meaningful but I wonder if `ps` wouldn't be enough?

File: src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st [hunk 7]
```diff
@@ -45,21 +45,30 @@ PhLImageKillCliCommand class >> launcherCmdPositionals [
...
 PhLImageKillCliCommand >> killAllImages [
-	 ^ LibC
-				runCommand:
-					'kill $(ps x | grep  -i Pharo | grep ''.image'' | tr -s '' '' | cut -d '' '' -f1)'
+	
+	^ LibC runCommand: (self killImageCmdStringFrom: nil)
+]
+
+{ #category : #commands }
+PhLImageKillCliCommand >> killImageCmdStringFrom: anImageName [
```
Comment (demarey) [inline src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st:59]: good idea!

File: src/PharoLauncher-CLI-Tests/PhLImageKillCliCommandTest.class.st [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+A PhLImageKillCliCommandTest is a test class for testing the behavior of PhLImageKillCliCommand
+"
+Class {
+	#name : #PhLImageKillCliCommandTest,
+	#superclass : #PhLImageCliCommandTest,
+	#category : #'PharoLauncher-CLI-Tests'
+}
+
+{ #category : #tests }
+PhLImageKillCliCommandTest >> testKillImageCmdStringFrom [
```
Comment (demarey) [inline src/PharoLauncher-CLI-Tests/PhLImageKillCliCommandTest.class.st:11]: Is this test really useful? It is so dependant on the implementation.

**Review by demarey — COMMENTED** [review]
Thanks for the contribution

File: test/testKillImage.sh (hunk 6 above)
Comment (Bajger) [reply test/testKillImage.sh]: I didn't know that (not familiar with Docker), for me it looked like bash specific command :) Is our assumption that users of this API would be similar user base as Docker users? I changed that because API uses English words also for other commends (kind of consistency). Let me know, I can revert this back, if you'd rather prefer `ps`.

Comment (demarey) [reply test/testKillImage.sh]: We can keep it. Users of this command will tell us if processList takes too much time to type or if they prefer a shorcut

File: src/PharoLauncher-CLI-Tests/PhLImageKillCliCommandTest.class.st (hunk 8 above)
Comment (Bajger) [reply src/PharoLauncher-CLI-Tests/PhLImageKillCliCommandTest.class.st:11]: I agree, probably it is too strict for checking exact content of this. Purpose of this test was to test if cascading of grep with image name parameter works correctly or not (and method is not failing). I will rewrite test case.

**Comment by Bajger** [comment]
@demarey : Hi Christophe! I've made these sh tests running locally, seems to be working after my changes. Please review and I think this branch can be integrated to dev branch. Next step is to run these tests on CI (as separate PR). I will create some additional issues, after playing with cli. Thanks in advance!

File: src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st (hunk 4 above)
Comment (Bajger) [reply src/PharoLauncher-CLI/PhLImageKillCliCommand.class.st:49]: done

**Review by Bajger — APPROVED** [review]
All sh unit tests should be working. Will add an invoking of theses tests via CI
